drizzlepac/alignimages.py

```diff
@@ -1036,96 +1084,129 @@ def generate_source_catalogs(imglist, **pars):
-                log.exit("ERROR! Unrecognized detector '{}'. Exiting...".format(detector))
-        else:
-            sys.error("ERROR! Unrecognized instrument '{}'. Exiting...".format(instrument))
-            log.exit("ERROR! Unrecognized instrument '{}'. Exiting...".format(instrument))
-
-        # Identify sources in image, convert coords from chip x, y form to reference WCS sky RA, Dec form.
-        imgwcs = HSTWCS(imghdu, 1)
-        # Convert fwhmpsf from arsec to pixels
-        fwhmpsf_pix = sourcecatalogdict[imgname]["params"]['fwhmpsf'] / imgwcs.pscale
-        sourcecatalogdict[imgname]["catalog_table"] = \
-            amutils.generate_source_catalog(imghdu, fwhm=fwhmpsf_pix, **detector_pars)
-
-        # write out coord lists to files for diagnostic purposes. Protip: To display the sources in these files in DS9,
-        # set the "Coordinate System" option to "Physical" when loading the region file.
-        imgroot = os.path.basename(imgname).split('_')[0]
-        num_sci = amutils.countExtn(imghdu)
-        # Allow user to decide when and how to write out catalogs to files
-        if output:
-            for chip in range(1, num_sci + 1):
-                chip_cat = sourcecatalogdict[imgname]["catalog_table"][chip]
-                if chip_cat and len(chip_cat) > 0:
-                    regfilename = "{}_sci{}_src.reg".format(imgroot, chip)
-                    out_table = Table(chip_cat)
-                    # To align with positions of sources in DS9/IRAF
-                    out_table['xcentroid'] += 1
-                    out_table['ycentroid'] += 1
-                    out_table.write(regfilename,
-                                    include_names=["xcentroid", "ycentroid"],
-                                    format="ascii.fast_commented_header")
-                    log.info("Wrote region file {}\n".format(regfilename))
-        imghdu.close()
-    return(sourcecatalogdict)
-
-
-# ----------------------------------------------------------------------------------------------------------------------
+        with fits.open(imgname) as f:
+            phdr = f[0].header
+            instrument = phdr['INSTRUME'].lower()
+            detector = phdr['DETECTOR'].lower()
+
+            # get instrument/detector-specific image alignment parameters
+            try:
+                detector_info = _DETECTOR_SPECIFIC_PARAMS[instrument]
+                try:
+                    detector_pars = detector_info[detector]
+                    # to allow generate_source_catalog to get detector
+                    # specific parameters
+                    detector_pars.update(pars)
+                    src_cat[imgname]["params"] = detector_pars
+
+                except KeyError:
+                    log.error(
+                        "ERROR! Unrecognized detector '{}'. Exiting..."
+                        .format(detector)
+                    )
+                    sys.exit(
```

Comment:
I am not really sure about the need to kill the whole Python **session** here (or anywhere else in a function).